### PR TITLE
LogAnalyser fixes

### LINKED ISSUE #2240: LogAnalyser: no longer working with ArduCopter 3.2.1 and ArduCopter 3.3-rc3
LogAnalyser seems to be broken with recent ArduCopter logs:

3.2.1:
```
Broken FMT message for STRT .. ignoring
Traceback (most recent call last):
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/LogAnalyzer.py", line 255, in <module>
    main()
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/LogAnalyzer.py", line 225, in main
    logdata = DataflashLog.DataflashLog(args.logfile.name, format=args.format, ignoreBadlines=args.skip_bad) # read log
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/DataflashLog.py", line 417, in __init__
    self.read(logfile, format, ignoreBadlines)
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/DataflashLog.py", line 462, in read
    numBytes, lineNumber = self.read_binary(f, ignoreBadlines)
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/DataflashLog.py", line 599, in read_binary
    for e in self._read_binary(f, ignoreBadlines):
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/DataflashLog.py", line 633, in _read_binary
    raise ValueError(str(h) + "unknown type")
ValueError: <logheader head1=0xa3 head2=0x95 msgid=0xa (10)>unknown type
```

3.3-dev:
```
../20150315-iris-Straach/23.BIN 
Broken FMT message for STRT .. ignoring
Traceback (most recent call last):
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/LogAnalyzer.py", line 255, in <module>
    main()
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/LogAnalyzer.py", line 225, in main
    logdata = DataflashLog.DataflashLog(args.logfile.name, format=args.format, ignoreBadlines=args.skip_bad) # read log
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/DataflashLog.py", line 417, in __init__
    self.read(logfile, format, ignoreBadlines)
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/DataflashLog.py", line 462, in read
    numBytes, lineNumber = self.read_binary(f, ignoreBadlines)
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/DataflashLog.py", line 605, in read_binary
    self.process(lineNumber, e)
  File "/home/holger-local/projects/uav/master/ardupilot/Tools/LogAnalyzer/DataflashLog.py", line 527, in process
    raise Exception("Unknown log type for MODE line {} {}".format(self.vehicleType, repr(e)))
Exception: Unknown log type for MODE line APM:Copter <Log__MODE TimeMS:10737 Mode:0 ModeNum:0>
```

### PR DESCRIPTION
**Pull request by hsteinhaus**
Fixes #2240. Seems to work again with ArduCopter 3.2.1 and ArduCopter 3.3-rc3 logs.

### CODE REVIEW

File: Tools/LogAnalyzer/DataflashLog.py [hunk 1]
```diff
@@ -521,7 +521,7 @@ def process(self, lineNumber, e):
                     self.modeChanges[lineNumber] = (modes[int(e.Mode)], e.ThrCrs)
                 except:
                     self.modeChanges[lineNumber] = (e.Mode, e.ThrCrs)
-            elif self.vehicleType == "ArduPlane" or self.vehicleType == "ArduRover":
+            elif self.vehicleType in ["ArduPlane", "APM:Plane", "ArduRover", "APM:Rover", "APM:Copter"]:
```
Comment (gmorph) [inline Tools/LogAnalyzer/DataflashLog.py]: I think APM:Copter should be in the entry above

File: Tools/LogAnalyzer/DataflashLog.py [hunk 2]
```diff
@@ -501,7 +501,7 @@ def process(self, lineNumber, e):
             else:
                 self.messages[lineNumber] = e.Message
         elif e.NAME == "MODE":
-            if self.vehicleType == "ArduCopter":
+            if self.vehicleType in ["ArduCopter"]:
```
Comment (gmorph) [inline Tools/LogAnalyzer/DataflashLog.py]: This should also include APM:Copter moved from below

**Comment by gmorph** [comment]
Hi Holger.  Can you make the simple fix I've outlined and re-submit the PR and we will merge it in.
Thanks, Grant.

File: Tools/LogAnalyzer/DataflashLog.py (hunk 1 above)
Comment (hsteinhaus) [inline Tools/LogAnalyzer/DataflashLog.py]: No, that wouldn't work. I moved it here intentionally to fix AttributeError execptions, as newer Copter releases write a mode line similar to Plane/Rover (no e.ThrCrs attribute, but e.Mode and e.ModeNum).

File: Tools/LogAnalyzer/DataflashLog.py (hunk 2 above)
Comment (hsteinhaus) [inline Tools/LogAnalyzer/DataflashLog.py]: That's indeed a remainder of trying to move in APM:Copter here (which did not work - see above). This particular change could be reverted therefore.

**Comment by gmorph** [comment]
Thanks Holger for updating.  We will get the change merged in.  Keep the code coming!
Thanks!

**Comment by rmackay9** [comment]
merged, thanks!

**Pull request closed without merging**